src/neptune/internal/operation_processors/async_operation_processor.py

```diff
@@ -162,7 +162,7 @@ def _check_no_progress(self) -> None:
 
         with self._lock:
             if self._should_call_no_progress_callback:
-                self._async_no_progress_callback()
+                threading.Thread(target=self._async_no_progress_callback, daemon=True).start()
```

Comment:
`daemon=True` False?

Reply:
Are you sure that Python doesn't require to call `join()`? I heard that sometimes it may be needed (it wasn't Python). Stackoverflow says it's fine: https://stackoverflow.com/questions/49912013/python-what-if-we-call-thread-start-and-leave-it-without-join-or-close but just to be sure :wink:

Reply:
> A daemon thread is a thread that runs in the background and does not prevent the program from exiting. In other words, if there are only daemon threads left in a Python program, the program will exit even if those threads are still running. Non-daemon threads, on the other hand, will keep the program alive until they complete their execution.

Reply:
With join the caller will wait and hold the lock indefinitely.

Reply:
I know the theory, all background jobs have `daemon=True`, and I think that callbacks shouldn't block the exit of program.

Reply:
> All background jobs have daemon=True, and I think that callbacks shouldn't block the exit of program.

We should probably ask someone from Product Team here :<

Reply:
> With join the caller will wait and hold the lock indefinitely.

Yeah, I know. I was thinking about joining at the end of a script but I've looked for something like that and it's fine that we're not calling join at all.

Reply:
But let's proceed with this fix. This could be changed anytime.